### ISSUE TITLE
Ensure we're not hitting any rate limits, and handle if we are

### ISSUE THREAD

**Issue by LeoMcA** [mozilla/discourse#63]
With the code written in https://github.com/mozilla/discourse/issues/13 an email sent to 20 users will send 20 queries to the Auth0 API. We should instead send one query, which returns the 20 users.

There's a few things blocking us here:

- <s>[Auth0 User Search doesn't work](https://community.auth0.com/questions/4916/user-search-with-multiple-user-ids-not-working)</s>
- <s>Some code in Discourse core could be restructured to allow easier monkey patching</s>

**Comment by LeoMcA** [comment]
https://github.com/LeoMcA/discourse-mozilla-iam/commit/237c1befa41d0340c6b672ed47831681cd00a31c

**Comment by LeoMcA** [comment]
> Some code in Discourse core could be restructured to allow easier monkey patching

This has been done: https://github.com/discourse/discourse/commit/ddc0134b4836c86dc4d56ef9e765f1bdc38fbc55

**Comment by LeoMcA** [comment]
**Sprint 12 status:**

Still blocked by Auth0. Waiting on reply to kang's support request.

**Comment by LeoMcA** [comment]
[Some code written](https://github.com/LeoMcA/discourse-mozilla-iam/tree/groups-sync-n+1), needs to be rewritten based on refactor of `discourse-mozilla-iam` and tested to see if it works/if it is more performant.

**Comment by LeoMcA** [comment]
Wontfix. Rate limiting will be a consideration while working on migration to Person API v2
